engine/collection/compliance/core.go

```diff
@@ -107,9 +120,29 @@ func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.Clus
 		Time("timestamp", header.Timestamp).
 		Hex("proposer", header.ProposerID[:]).
 		Hex("signers", header.ParentVoterIndices).
+		Uint64("final_height", finalHeight).
 		Logger()
+	if log.Debug().Enabled() {
+		log = log.With().Strs("tx_ids", flow.IdentifierList(proposal.Payload.Collection.Light().Transactions).Strings()).Logger()
+	}
 	log.Info().Msg("block proposal received")
 
+	// drop proposals below the finalized threshold
+	if header.Height < finalHeight || header.View < finalView {
```

Comment:
I think we could also drop a proposal if its height is equal to finalized height
```suggestion
	if header.Height <= finalHeight || header.View <= finalView {
```